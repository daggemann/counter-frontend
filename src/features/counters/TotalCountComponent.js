import React from 'react';
import styled from 'styled-components';

const TotalCount = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    margin-top: 10px;
`;

const Text = styled.span``;
Text.displayName = 'Text';

const Total = styled.span`
    margin-left: 10px;
`;
Total.displayName = 'Total';

export default function TotalCountComponent(props) {
    return (
        <TotalCount>
            <Text>Total:</Text>
            <Total>{props.total || 0}</Total>
        </TotalCount>
)
}