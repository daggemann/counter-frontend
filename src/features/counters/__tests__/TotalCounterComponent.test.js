import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TotalCountComponent from "../TotalCountComponent";

test('render total count', () => {
    const tree = renderer.create(<TotalCountComponent total={10}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ensure that the total is equal to passed prop', () => {
    const wrapper = shallow(<TotalCountComponent total={10}/>)
    const total = wrapper.find('Total');
    expect(total.get(0).props.children).toEqual(10);
});