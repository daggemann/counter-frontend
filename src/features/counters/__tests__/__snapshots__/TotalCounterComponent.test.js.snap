// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render total count 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-top: 10px;
}

.c1 {
  margin-left: 10px;
}

<div
  className="c0"
>
  <span
    className=""
  >
    Total:
  </span>
  <span
    className="c1"
  >
    10
  </span>
</div>
`;
